driver/ssh/config.go

```diff
@@ -47,13 +52,15 @@ func (conf *Config) DecodeFromDSN(encodedDSN string) (dsn string, err error) {
 
 	parsed.RawQuery = values.Encode()
 
-	// remove the middle information of
-	// scheme://ssh_user:ssh_password@ssh_host:ssh_port/
+	// remove the middle information of scheme://ssh_user:ssh_password@ssh_host:ssh_port/
 	splitted := strings.Split(parsed.String(), "://")
-	idx := strings.Index(splitted[1], "/")
+	if len(splitted) < 2 {
+		return "", errors.New("invalid DSN format to read the warehouse DSN: missing scheme")
+	}
 
+	idx := strings.Index(splitted[1], "/")
 	if idx == -1 {
-		return "", fmt.Errorf("fetching index of / to start reading the warehouse dsn: %s", err.Error())
+		return "", errors.New("invalid DSN format to read the warehouse DSN: missing host and port")
```

Comment:
Same comment about `warehouse` here.

```suggestion
		return "", fmt.Errorf("invalid DSN format: missing host and port")
```